workloads/templates/common-templates.md

```diff
@@ -131,3 +131,46 @@ oc process rhel7-generic-tiny PVCNAME=mydisk NAME=rheltinyvm | oc apply -f -
 
 Please note that, after the generation step, VM objects and template objects have no relationship with each other besides the aforementioned label (e.g. changes
 in templates do not automatically affect VMs, or vice versa).
+
+## common template customization
```

Comment:
The section needs to describe what you can do to the template, which special annotations we have there, what labels are provided and what the meanings are. Creating a VM from the template can be in separate section if you wish, but we need to describe the format of the template.

Reply:
OK, this means we should document most if not all[1] the annotations listed here: https://github.com/fromanirh/common-templates/blob/template-validation-docs/templates/README.md#templateyaml

this will duplicate some information between documents, but it looks like a necessary evil - AFAICT user doc shouldn't directly link developer doc, even thoug lines are a bit blurred for templates.

I'll also add a couple of "oc process patch" command examples only, being it strightforward to do for the other annotations.

+++

[1] for example the version field should be pretty self explanatory, maybe just a big red label "don't touch this"